server/src/main/java/org/opensearch/telemetry/tracing/listener/TraceableSearchRequestOperationsListener.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing.listener;
+
+import org.opensearch.action.search.SearchPhaseContext;
+import org.opensearch.action.search.SearchRequestContext;
+import org.opensearch.action.search.SearchRequestOperationsListener;
+import org.opensearch.telemetry.tracing.AttributeNames;
+import org.opensearch.telemetry.tracing.Span;
+import org.opensearch.telemetry.tracing.SpanBuilder;
+import org.opensearch.telemetry.tracing.SpanContext;
+import org.opensearch.telemetry.tracing.SpanScope;
+import org.opensearch.telemetry.tracing.Tracer;
+import org.opensearch.telemetry.tracing.noop.NoopSpan;
+
+import static org.opensearch.core.common.Strings.capitalize;
+
+/**
+ * SearchRequestOperationsListener subscriber for search request tracing
+ *
+ * @opensearch.internal
+ */
+public final class TraceableSearchRequestOperationsListener extends SearchRequestOperationsListener {
+    private final Tracer tracer;
+    private final Span requestSpan;
+    private Span phaseSpan;
+    private SpanScope phaseSpanScope;
+
+    public TraceableSearchRequestOperationsListener(final Tracer tracer, final Span requestSpan) {
+        this.tracer = tracer;
+        this.requestSpan = requestSpan;
+        this.phaseSpan = NoopSpan.INSTANCE;
+    }
+
+    @Override
+    protected void onPhaseStart(SearchPhaseContext context) {
+        phaseSpan = tracer.startSpan(
+            SpanBuilder.from("coord" + capitalize(context.getCurrentPhase().getName()), new SpanContext(requestSpan))
+        );
+        phaseSpanScope = tracer.withSpanInScope(phaseSpan);
+    }
+
+    @Override
+    protected void onPhaseEnd(SearchPhaseContext context, SearchRequestContext searchRequestContext) {
+        phaseSpan.endSpan();
+        phaseSpanScope.close();
+    }
+
+    @Override
+    protected void onPhaseFailure(SearchPhaseContext context) {
```

Comment:
This is incompleteness of `SearchRequestOperationsListener`, we need the cause of the failure:

 @Override
    protected void onPhaseFailure(SearchPhaseContext context, , Throwable cause) {   
        phaseSpan.setError(cause);
        phaseSpan.endSpan();
        phaseSpanScope.close();
   }

Reply:
Updated all `onPhaseFailure()` with `Throwable cause` parameter